### ISSUE TITLE
weird month-ing

### ISSUE THREAD

**Issue by maximburdiean** [Alexpol19/AdminUi#6]
Для определения названия месяца, определена константа и название определяется по индексу месяца из даты:
![image](https://user-images.githubusercontent.com/5040836/98916482-e4304480-24d3-11eb-8efb-3e5caa3b2020.png)

**Comment by Alexpol19** [comment]
Не понимаю! Как можно иначе?
Мне нужны были названия месяца в сокращенном варианте поэтому создал отдельно

**Comment by maximburdiean** [comment]
Андрей, обсудите, пожалуйста, с Алексеем альтернативы

**Comment by aandrey1993** [comment]
Добрый вечер, Алексей.

Попробуйте посмотреть в сторону библиотеки `moment js`. Данная библиотека позволяет форматировать даты любым способом, вплоть до определения названия месяцев. Также для данной библиотеки можно настроить `locale` и получить название сразу на нужном языке. Ссылка на библиотеку: https://momentjs.com . На оф. сайте есть масса примеров. Попробуйте подключить данную библиотеку к проекту и отформатировать даты с её помощь, чтобы уйти от массива `months `

P.S. Форматирование дат лучше вынести из `render` функции. Проще создать отдельную функцию в данном компоненте для этого, которая будет вызываться из `render` функции для `startDate` и `endDate`

**Comment by Alexpol19** [comment]
@aandrey1993 Понял. Попробую исправить с помощью moment.js. Сталкивался уже но не вникал с этой библиотекой

**Comment by Alexpol19** [comment]
Исправил с помощью Moment.js. Также попробовал сменить язык - тоже получилось( Оставил закомментированным )

**Comment by aandrey1993** [comment]
Сделали commit исправлений, чтобы можно посмотреть что получилось ?

**Comment by Alexpol19** [comment]
@aandrey1993 Еще 1 задача осталась(https://github.com/Alexpol19/AdminUi/issues/4#issue-741400003). Работаю и не особо есть время отвлекаться. Постараюсь до вечера сделать

**Comment by Alexpol19** [comment]
@aandrey1993 Сделал commit исправлений. Можете посмотреть

**Comment by aandrey1993** [comment]
С применением moment js, работа с датами стало понятнее. Код для компонента `MyChart` можно еще упростить. Функцию `formatDate`, изменить для форматирования одной даты. Вместо применения конкатенации строк, лучше использовать [шаблонные  строки](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/template_strings).  Например:

![изображение](https://user-images.githubusercontent.com/49781313/99105136-abcf5a00-25ea-11eb-9c9a-e8c22a7d3a12.png)

![изображение](https://user-images.githubusercontent.com/49781313/99105201-c570a180-25ea-11eb-8ce1-9919dfaa7229.png)

**Comment by aandrey1993** [comment]
@Alexpol19, Подскажите в чем необходимость даты из `DateRangePicker` дополнительно оборачивать в объект `Date` ? (Компонент `DateRangeChart`)

![изображение](https://user-images.githubusercontent.com/49781313/99106539-0e295a00-25ed-11eb-8ee6-e28f3c7e40b5.png)

**Comment by Alexpol19** [comment]
@aandrey1993 Честно даже не знаю. Может я таким образом пытался разделить общий range на 2 переменные, чтоб было понятнее когда не использовал moment, чтоб конечный вывод был покороче( без range.startDate.getDate()....)
Но на самом деле они одинаковые так что это лишний код

**Comment by aandrey1993** [comment]
Да, но разделить на переменные можно без вызова конструктора Date. Например:

```javascript
const startDate = range.startDate
```

`DateRangePicker` возвращает range уже в Date формате

**Comment by Alexpol19** [comment]
Знаю знаю. Полностью согласен. Не подумал.

**Comment by Alexpol19** [comment]
Можно вообще без переменных лишних
